mmv1/templates/terraform/examples/apphub_service_full.tf.erb

```diff
@@ -0,0 +1,113 @@
+resource "google_apphub_application" "application" {
+  location = "us-east1"
+  application_id = "<%= ctx[:vars]['application_id'] %>"
+  scope {
+    type = "REGIONAL"
+  }
+}
+
+resource "google_project" "service_project" {
+  project_id ="<%= ctx[:vars]['service_project_attachment_id'] %>"
+  name = "Service Project"
+  org_id = "<%= ctx[:test_env_vars]['org_id'] %>"
+}
+
+resource "google_apphub_service_project_attachment" "service_project_attachment" {
+  service_project_attachment_id = google_project.service_project.project_id
+}
+
+# discovered service block
+data "google_apphub_discovered_service" "catalog-service" {
+  provider = google
+  location = "us-east1"
+  # ServiceReference | Application Hub | Google Cloud
+  # Using this reference means that this resource will not be provisioned until the forwarding rule is fully created
+  service_uri = "//compute.googleapis.com/${google_compute_forwarding_rule.forwarding_rule.id}"
+  depends_on = [google_apphub_service_project_attachment.service_project_attachment]
+}
+
+resource "google_apphub_service" "<%= ctx[:primary_resource_id] %>" {
+  location = "us-east1"
+  applications_id = google_apphub_application.application.application_id
+  service_id = google_compute_forwarding_rule.forwarding_rule.name
+  discovered_service = data.google_apphub_discovered_service.catalog-service.discovered_service[0].name
```

Comment:
What format does the discovered_service.name have here, and why do we need to go through this data source to retrieve it? Is it different than either the service_uri or the ID of the Terraform resource?

Reply:
The format for the same is projects/{HOST_PROJECT_ID}/locations/{LOCATION}/discoveredServices/{DISCOVERED_SERVICE_ID}
Yes, this is different than service_uri or id as this is a new resource being created by Apphub. There is no other way to get the name field as it is a read-only resource.